pytket/phir/rebasing/rebaser.py

```diff
@@ -8,30 +8,24 @@
 
 from typing import TYPE_CHECKING
 
-from pytket.extensions.quantinuum.backends.api_wrappers import QuantinuumAPIOffline
-from pytket.extensions.quantinuum.backends.quantinuum import (
-    QuantinuumBackend,
-)
+from pytket.circuit import OpType
 from pytket.passes import DecomposeBoxes
+from pytket.passes.auto_rebase import auto_rebase_pass
+from pytket.phir.qtm_machine import QTM_MACHINES_MAP, QtmMachine
 
 if TYPE_CHECKING:
     from pytket.circuit import Circuit
 
 
-def rebase_to_qtm_machine(
-    circuit: "Circuit", qtm_machine: str, tket_optimization_level: int
-) -> "Circuit":
+def rebase_to_qtm_machine(circuit: "Circuit", qtm_machine: QtmMachine) -> "Circuit":
     """Rebases a circuit's gate to the gate set appropriate for the given machine."""
-    qapi_offline = QuantinuumAPIOffline()
-    backend = QuantinuumBackend(
-        device_name=qtm_machine,
-        machine_debug=False,
-        api_handler=qapi_offline,  # type: ignore [arg-type]
+    machine = QTM_MACHINES_MAP.get(qtm_machine)
+    gateset = (
+        {OpType.Rz, OpType.PhasedX, OpType.ZZPhase}
```

Comment:
This is roughly equivalent to the Quantinuum gateset, do we want to rebase to that in the case of no machine - seems like it might reduce the flexibility of allowing this to be run without any Quantinuum specificity.

Reply:
I changed the code to use `QTM_DEFAULT_GATESET` explicitly. I don't think this reduces flexibility, as one can specify a machine with another gateset if desired.